testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -537,6 +537,28 @@ public void testMaterializedView()
                         "FROM\n" +
                         "  nation");
 
+        assertQueryFails(
+                format("ALTER TABLE %s RENAME TO %s_new", viewName, viewName),
+                format("line 1:1: Table '.*.%s' does not exist. Attempting to use table DDL syntax on a materialized view.", viewName));
+
+        assertUpdate(format("ALTER TABLE IF EXISTS %s RENAME TO %s_new", viewName, viewName));
+
+        assertQueryFails(
```

Comment:
does it still behave same way if connector supports materialized views but not views (not sure if we have a case like that now)

Reply:
I think we don't have a case like that. The only connector that supports materialized views is the iceberg connector and it also supports views.